File: 2559_count_vowel_strings_in_ranges.cpp
class Solution {
public:
    vector<int> vowelStrings(vector<string>& x1, vector<vector<int>>& x2) {
        const unsigned b=1+(1<<4)+(1<<8)+(1<<14)+(1<<20);
        int p=x1.size(), u=x2.size();
        vector<int> d(p+1, 0);
        for(int k=0; k<p; k++)
            d[k+1]=d[k]+(((1<<(x1[k][0]-'a'))&b)&&((1<<(x1[k].back()-'a'))&b));
        vector<int> z(u, 0);
        for(int v=0; v<u; v++)
            z[v]=d[x2[v][1]+1]-d[x2[v][0]];
        return z;
    }
};

File: 2559_count_vowel_strings_in_ranges.java
class Solution {
    public int[] vowelStrings(String[] w, int[][] q) {
        final int v = 1 + (1 << ('e' - 'a')) + (1 << ('i' - 'a')) + (1 << ('o' - 'a')) + (1 << ('u' - 'a'));
        int n = w.length, m = q.length;
        int[] c = new int[n + 1];
        for (int i = 0; i < n; i++) 
            c[i + 1] = c[i] + (((1 << (w[i].charAt(0) - 'a')) & v) > 0 && ((1 << (w[i].charAt(w[i].length() - 1) - 'a')) & v) > 0 ? 1 : 0);
        int[] r = new int[m];
        for (int i = 0; i < m; i++) 
            r[i] = c[q[i][1] + 1] - c[q[i][0]];
        return r;
    }
}
